nxp: imx8ulp: add support for and enable pinctrl

LaurentiuM1234 opened the pull request:
Depends on:

- [x] https://github.com/zephyrproject-rtos/zephyr/pull/70219

LaurentiuM1234 committed:
nxp: imx8ulp: enable pinctrl

This commit enables pinctrl on i.MX8ULP. This includes:
	1) Adding `pinctrl_soc.h` header file.
	2) Adding DTS node for IOMUXC1, which is one of the
	IPs responsible for managing the 8ULP pads.
	3) Adding .dtsi with pin definitions. For now, only
	the LPUART7 pads are added to this file because this
	is going to be the only consummer for now.
	4) Modifying the `pinctrl_imx.c` driver to work for 8ULP.
	5) Enabling the `CONFIG_HAS_MCUX_IOMUXC`, which is a
	dependency of `CONFIG_PINCTRL_IMX`.

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>

LaurentiuM1234 committed:
drivers: pinctrl: Kconfig.imx: enable PINCTRL_IMX by default

Currently, when using the `pinctrl_imx.c` driver one has
to manually enable `CONFIG_PINCTRL_IMX` even if the pinctrl
node using the driver is enabled. This is redundant and prone
to error as one would expect the driver to be enabled when
the node is also enabled.

This commit fixes the issue by enabling `CONFIG_PINCTRL_IMX`
by default and adding a new dependency on the state of the
pinctrl node (i.e: node needs to be enabled for the configuration
to also be set to `y`).

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>

LaurentiuM1234 commented:
@gmarull pls review

gmarull commented on dts/bindings/pinctrl/nxp,imx8ulp-pinctrl.yaml:
```diff
@@ -0,0 +1,36 @@
+          this property will be used to configure the pad for high
+          drive strength. If not present, the standard drive strength
+          will be used.
+      slew-rate:
+        type: boolean
+        description: |
```
this property is misleading as a boolean, please, make it a choice between standard/slow (also, explain what "standard" means...)

LaurentiuM1234 replied:
Thanks! Switched to `string` type + enum. Also changed comments for `slew-rate` and `drive-strength`. Should be more useful now.

LaurentiuM1234 commented:
**V2 changes**

1. Changed type of `slew-rate` and `drive-strength` properties to `string`. Also, modified comments to make the properties more clear.
2. Rebased.
